discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -486,9 +489,12 @@ def prepare_for_install(
                 static_ips=static_ips_config
             )
         nodes.start_all()
-        self.wait_until_hosts_are_discovered(nodes_count=nodes_count)
+        self.wait_until_hosts_are_discovered(nodes_count=nodes_count, allow_insufficient=True)
```

Comment:
What's the purpose of adding `allow_insufficient=True` ? Could this be moved to be a parameter to `prepare_for_install` ?

Reply:
I think we should always allow it.  I don't know why it is not always wait for insufficient as well.
I don't think the user of prepare_for_install should bother with this implementation detail

Reply:
In the majority of cases `wait_until_hosts_are_discovered` assumes hosts are either known or pending-for-input
an insufficient host is the exception

Reply:
Waiting for insufficient is more correct.

Reply:
Sure, it doesn't break any flow as prepare_for_install still waits for cluster to be ready

Reply:
I don't think it should break anything